### PR TITLE
Stronger ESLint (no longer rely on TSDX for linting)

### PR DESCRIPTION
**Pull request by wmai**
The purpose of this PR fulfils two objectives:
- strengthen the linting and type check of the `api-client` and `visualizations` packages
- to start getting rid of **TSDX** which is no longer maintained

So via this PR we won't execute `tsdx lint` anymore but directly the commands of **ESLint** and **Prettier**. A small step for a developper, and soon, a giant leap for the codebase.

The commits containing the modifications made on the configuration files are those there:
- [8695eff](https://github.com/opendatasoft/ods-dataviz-sdk/pull/103/commits/8695eff28d293e2fbc067294ad1dda985b2e5f23)
- [ff7c944](https://github.com/opendatasoft/ods-dataviz-sdk/pull/103/commits/ff7c9444597515a9296fe9f6f25022c81ed80033)

And if you have the courage to review it, here is the commit containing the linting error repairs: [63e5a1b](https://github.com/opendatasoft/ods-dataviz-sdk/pull/103/commits/63e5a1bfc7bd4b77c5087a30a5ec02d19dd2c157)

The `visualizations-react` package has been deliberately left out because it requires a lot of repair in many `*.stories.ts[x]` files after configuration changes.

## To test
- That the linter config works well with your IDE
- That the compiled output works well when linked to `platform`

### CODE REVIEW

**Review by RafaelSzmarowski — APPROVED** [review]
LGTM ! Nice work ! 

As a reminder, even if it's not blocking for now, I have this warning in my console :  

```
Cannot find type definition file for 'jest'.
The file is in the program because:
Entry point of type library 'jest' specified in compilerOptions
```

**Review by etienneburdet — CHANGES_REQUESTED** [review]
Thank you for doing this, it's a real chore… 💪 It seems to be mostly working, although I don't have eslint error in Svelte files in my ide (but they are corrected on save…). I'll work on that.

My main suggestion would be to try to factor as much as we can for dev config. I think we could have a top level tsconfig with [this technique](https://blog.logrocket.com/setting-up-monorepo-with-lerna-typescript/), we can have a base eslintconfig that we import in each package and then extend (most of it is shared), move all [the tasks (lint, tests etc.) to top level](https://lerna.js.org/docs/faq#root-packagejson) and maybe use hoisting for devDependencies. What do you think?

File: packages/api-client/test/client-node.test.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 /**
  * @jest-environment node
  */
+
+/* eslint-disable no-new */
```
[the remark this answers is not in the document]
Comment (wmai) [reply packages/api-client/test/client-node.test.ts:5]: Actually I think we want to catch the error when an instance of ApiClient is created here. If we create the instance outside the `expect` block, we won't be able to catch the error.

[the remark this answers is not in the document]
Comment (wmai) [reply packages/api-client/test/client-node.test.ts:5]: Actually I think we want to catch the error when an instance of ApiClient is created here. If we create the instance outside the `expect` block, we won't be able to catch the error.

File: packages/api-client/package.json [hunk 2]
```diff
@@ -30,8 +30,12 @@
...
         "test": "tsdx test",
-        "lint": "tsdx lint src test",
-        "lint:fix": "tsdx lint src test --fix",
+        "lint": "run-p lint:*:check -c",
+        "lint:fix": "run-p lint:*:fix -c",
+        "lint:eslint:check": "eslint src test",
+        "lint:prettier:check": "prettier --check src test",
+        "lint:eslint:fix": "eslint src test --fix",
+        "lint:prettier:fix": "prettier src test --write",
         "prepare": "tsdx build",
```
[the remark this answers is not in the document]
Comment (wmai) [reply packages/api-client/package.json:39]: I'm sorry there must be something I didn't understand about Lerna. Each of these scripts (although named the same) are different in each package. How would you move them to the parent `package.json`?

**Comment by wmai** [comment]
Thank you very much for the thorough review @etienneburdet 🙏

> My main suggestion would be to try to factor as much as we can for dev config. I think we could have a top level tsconfig with [this technique](https://blog.logrocket.com/setting-up-monorepo-with-lerna-typescript/), we can have a base eslintconfig that we import in each package and then extend (most of it is shared), move all [the tasks (lint, tests etc.) to top level](https://lerna.js.org/docs/faq#root-packagejson) and maybe use hoisting for devDependencies. What do you think?

Yes, I also wanted to have a common `tsconfig` configuration because the different packages share many properties. But the problem is that you can only inherit one file via the `extends` parameter, and the `visualizations` package already inherits ``@tsconfig/svelte/tsconfig.json``.

And regarding the ESLint configuration sharing, I think it would indeed be nice to try to factor that. But we will have to get rid of TSDX in the `visualizations-react` package first.

**Comment by etienneburdet** [comment]
👌However, thinking about it, we might want to have a separate PR for the factoring. First remove tsdx one package at a time, then be smart 😂

File: packages/api-client/test/client-node.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (etienneburdet) [reply packages/api-client/test/client-node.test.ts:5]: ha yes, and const xxx = new ApiClient, will raise an "unused". Not worth it indeed…

**Pull request merged**